bittensor/commands/weights.py

```diff
@@ -0,0 +1,257 @@
+            if "subtensor" in locals():
+                subtensor.close()
+                bittensor.logging.debug("closing subtensor connection")
+
+    @staticmethod
+    def _run(cli: "bittensor.cli", subtensor: "bittensor.subtensor"):
+        r"""Commit weights for a specific subnet"""
+        wallet = bittensor.wallet(config=cli.config)
+
+        # Get values if not set
+        if not cli.config.is_set("netuid"):
+            cli.config.netuid = int(Prompt.ask(f"Enter netuid"))
+
+        if not cli.config.is_set("uids"):
+            cli.config.uids = Prompt.ask(f"Enter UIDs (comma-separated)")
+
+        if not cli.config.is_set("weights"):
+            cli.config.weights = Prompt.ask(f"Enter weights (comma-separated)")
+
+        # Parse from string
+        netuid = cli.config.netuid
+        uids = np.array(
+            list(map(int, re.split(r"[ ,]+", cli.config.uids))), dtype=np.int64
+        )
+        weights = np.array(
+            list(map(float, re.split(r"[ ,]+", cli.config.weights))), dtype=np.float32
+        )
+        weight_uids, weight_vals = weight_utils.convert_weights_and_uids_for_emit(
+            uids=uids, weights=weights
+        )
+
+        # Run the commit weights operation
+        success, message = subtensor.commit_weights(
+            wallet=wallet,
+            netuid=netuid,
+            uids=weight_uids,
+            weights=weight_vals,
+            wait_for_inclusion=cli.config.wait_for_inclusion,
+            wait_for_finalization=cli.config.wait_for_finalization,
+            prompt=cli.config.prompt,
+        )
+
+        # Result
+        if success:
+            bittensor.__console__.print(f"Weights committed successfully")
+        else:
+            bittensor.__console__.print(f"Failed to commit weights: {message}")
+
+    @staticmethod
+    def add_args(parser: argparse.ArgumentParser):
+        parser = parser.add_parser(
+            "commit", help="""Commit weights for a specific subnet."""
+        )
+        parser.add_argument("--netuid", dest="netuid", type=int, required=False)
+        parser.add_argument("--uids", dest="uids", type=str, required=False)
+        parser.add_argument("--weights", dest="weights", type=str, required=False)
+        parser.add_argument(
+            "--wait-for-inclusion",
+            dest="wait_for_inclusion",
+            action="store_true",
+            default=False,
+        )
+        parser.add_argument(
+            "--wait-for-finalization",
+            dest="wait_for_finalization",
+            action="store_true",
+            default=True,
+        )
+        parser.add_argument(
+            "--prompt",
+            dest="prompt",
+            action="store_true",
+            default=False,
+        )
+
+        bittensor.wallet.add_args(parser)
+        bittensor.subtensor.add_args(parser)
+
+    @staticmethod
+    def check_config(config: "bittensor.config"):
+        if not config.is_set("wallet.name") and not config.no_prompt:
```

Comment:
I would suggest swapping the terms. Since Python checks from left to right, it is better to start with lightweight calculations or with parts that have already been calculated.
`if not config.no_prompt and not config.is_set("wallet.name"):`
The same below.